src/Resources/public/Admin.js

```diff
@@ -322,7 +322,19 @@ var Admin = {
         });
 
         jQuery('.sonata-filter-form', subject).on('submit', function () {
-            jQuery(this).find('[sonata-filter="true"]:hidden :input').val('');
+            var $form = jQuery(this);
+            $form.find('[sonata-filter="true"]:hidden :input').val('');
+
+            var defaultValues = $.param({'filter': JSON.parse(this.dataset.defaultValues)}).split('&');
+            var submittedValues = $form.serialize().split('&');
+
+            var diffValues = submittedValues.filter(function (keyValueString) {
+                return defaultValues.indexOf(keyValueString) === -1 && keyValueString.split('=')[1] !== '';
+            });
+
+            window.location = this.action + (diffValues.length ? ('?' + diffValues.join('&')) : '');
+
+            return false;
```

Comment:
I'm not updated with Javascript, but last time I checked (many years ago), `return false;` was not a best practice. IMO we should keep the "normal" flow of the form and do not use `window.location`.

Reply:
Im dont view something bad in it.
You can find other usages in your project(also Admin.js) with `window.location =`

Reply:
Hmm. Only read access. Ok. What is your suggeston to refactor it?

Im will see:
1) Disable no need to submit values.
2) Create fake form with this url and submit it.